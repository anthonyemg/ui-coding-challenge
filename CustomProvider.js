import React from 'react';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import reducer from './src/public/reducers';

const store = createStore(reducer);

const CustomProvider = ({ children }) => {
  return <Provider store={store}>{children}</Provider>;
};

export default CustomProvider;
