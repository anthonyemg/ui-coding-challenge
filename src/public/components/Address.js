import React from 'react';
import { connect } from 'react-redux';
import { updateAddress, updateAddressLine1, showAddressModal, hideAddressModal } from '../actions';
import Modal from './Modal';
import AddressModal from './AddressModal';

class Address extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      updatedAddressLine1: '',
      updatedAddressLine2: ''
    };
    this.handleUpdateAddressLine1 = this.handleUpdateAddressLine1.bind(this);
    this.handleUpdateAddressLine2 = this.handleUpdateAddressLine2.bind(this);
    this.handleSubmitUpdatedAddress = this.handleSubmitUpdatedAddress.bind(this);
    this.handleCancel = this.handleCancel.bind(this);
  }
  handleUpdateAddressLine1(e) {
    this.setState({
      updatedAddressLine1: e.target.value
    });
  }
  handleUpdateAddressLine2(e) {
    this.setState({
      updatedAddressLine2: e.target.value
    });
  }
  handleSubmitUpdatedAddress() {
    this.props.updateAddress({
      line1: this.state.updatedAddressLine1,
      line2: this.state.updatedAddressLine2
    });
  }
  handleCancel() {
    this.setState({
      updatedAddressLine1: this.props.address.line1,
      updatedAddressLine2: this.props.address.line2
    });
  }
  render() {
    return (
      <div className="component-container">
        <div className="component-wrapper">
          <div className="component-details">
            <span className="component-title">Address</span>
            <span className="component-text">{this.props.address.line1}</span>
            <span className="component-text">{this.props.address.line2}</span>
          </div>
          <div>
            <button className="component-editButton" onClick={() => this.props.showAddressModal()}>
              Edit Address
            </button>
          </div>
        </div>
        {this.props.displayAddressModal ? (
          <Modal>
            <AddressModal
              address={this.props.address}
              updateAddress={this.props.updateAddress}
              handleUpdateAddress={this.handleUpdateAddress}
              handleUpdateAddressLine1={this.handleUpdateAddressLine1}
              handleUpdateAddressLine2={this.handleUpdateAddressLine2}
              handleSubmitUpdatedAddress={this.handleSubmitUpdatedAddress}
              hideAddressModal={this.props.hideAddressModal}
              handleCancel={this.handleCancel}
            />
          </Modal>
        ) : null}
      </div>
    );
  }
}

const mapStateToProps = state => ({
  address: state.address,
  displayAddressModal: state.displayAddressModal
});
const mapDispatchToProps = dispatch => ({
  updateAddress: address => dispatch(updateAddress(address)),
  showAddressModal: () => dispatch(showAddressModal()),
  hideAddressModal: () => dispatch(hideAddressModal())
});

export default connect(mapStateToProps, mapDispatchToProps)(Address);
