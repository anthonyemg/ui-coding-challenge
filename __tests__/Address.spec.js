import React from 'react';
import renderer from 'react-test-renderer';
import CustomProvider from '../CustomProvider';
import Address from '../src/public/components/Address';

test('the Address component renders correctly', () => {
  const component = renderer.create(
    <CustomProvider>
      <Address />
    </CustomProvider>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
