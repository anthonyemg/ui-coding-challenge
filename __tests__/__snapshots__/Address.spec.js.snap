// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Address component renders correctly 1`] = `
<div
  className="component-container"
>
  <div
    className="component-wrapper"
  >
    <div
      className="component-details"
    >
      <span
        className="component-title"
      >
        Address
      </span>
      <span
        className="component-text"
      >
        
      </span>
      <span
        className="component-text"
      >
        
      </span>
    </div>
    <div>
      <button
        className="component-editButton"
        onClick={[Function]}
      >
        Edit Address
      </button>
    </div>
  </div>
</div>
`;
